File: p2p/src/main/java/com/androdevlinux/percy/p2p/common/listeners/ServiceDiscoveredListener.java
package com.androdevlinux.percy.p2p.common.listeners;

import com.androdevlinux.percy.p2p.common.P2PDeviceService;
import com.androdevlinux.percy.p2p.common.WiFiP2PError;

import java.util.List;

public interface ServiceDiscoveredListener {

    void onNewServiceDeviceDiscovered(P2PDeviceService p2PDeviceService);

    void onFinishServiceDeviceDiscovered(List<P2PDeviceService> p2PDeviceServices);

    void onError(WiFiP2PError wiFiP2PError);

}

File: p2p/src/main/java/com/androdevlinux/percy/p2p/common/listeners/ServiceDiscoveredListener.kt
package com.androdevlinux.percy.p2p.common.listeners

import com.androdevlinux.percy.p2p.common.P2PDeviceService
import com.androdevlinux.percy.p2p.common.WiFiP2PError

interface ServiceDiscoveredListener {

    fun onNewServiceDeviceDiscovered(p2PDeviceService: P2PDeviceService)

    fun onFinishServiceDeviceDiscovered(p2PDeviceServices: List<P2PDeviceService>)

    fun onError(wiFiP2PError: WiFiP2PError)

}
